packages/core/application/core-application-browser-internal/src/utils/get_app_info.ts

```diff
@@ -7,48 +7,37 @@
  */
 
 import {
-  AppNavLinkStatus,
-  AppStatus,
   type App,
+  AppStatus,
   type AppDeepLink,
   type PublicAppInfo,
   type PublicAppDeepLinkInfo,
+  type AppDeepLinkLocations,
 } from '@kbn/core-application-browser';
 
+export const linkVisibleEverywhere: AppDeepLinkLocations[] = ['globalSearch', 'sideNav'];
+
 export function getAppInfo(app: App): PublicAppInfo {
-  const { updater$, mount, navLinkStatus = AppNavLinkStatus.default, ...infos } = app;
+  const { updater$, mount, visibleIn = linkVisibleEverywhere, ...infos } = app;
   return {
     ...infos,
-    status: app.status!,
-    navLinkStatus:
-      navLinkStatus === AppNavLinkStatus.default
-        ? app.status === AppStatus.inaccessible
-          ? AppNavLinkStatus.hidden
-          : AppNavLinkStatus.visible
-        : navLinkStatus,
-    searchable:
-      app.searchable ??
-      (navLinkStatus === AppNavLinkStatus.default || navLinkStatus === AppNavLinkStatus.visible),
+    status: app.status ?? AppStatus.accessible,
+    visibleIn: app.status === AppStatus.inaccessible ? [] : visibleIn,
     appRoute: app.appRoute!,
     keywords: app.keywords ?? [],
-    deepLinks: getDeepLinkInfos(app.deepLinks),
+    deepLinks: app.status === AppStatus.inaccessible ? [] : getDeepLinkInfos(app.deepLinks),
   };
 }
 
 function getDeepLinkInfos(deepLinks?: AppDeepLink[]): PublicAppDeepLinkInfo[] {
   if (!deepLinks) return [];
 
   return deepLinks.map(
-    ({ navLinkStatus = AppNavLinkStatus.default, ...rawDeepLink }): PublicAppDeepLinkInfo => {
+    ({ visibleIn = ['globalSearch'], ...rawDeepLink }): PublicAppDeepLinkInfo => {
```

Comment:
Should we have a common const value for `'[globalSearch']`? Perhaps `DEFAULT_LINK_VISIBILITY` and share with consumers? Might be useful if this ever changes (like a new place gets added)

Reply:
Good idea 👍 Done in https://github.com/elastic/kibana/pull/176383/commits/6ea6e0ff2928584b97f677a3f270d1d78f95fd4a